sdk/scheduler/src/test/java/com/mesosphere/sdk/specification/DefaultPlanGeneratorTest.java

```diff
@@ -66,57 +68,61 @@ public void testFullManualPlan() throws Exception {
         for (Map.Entry<String, RawPlan> entry : rawServiceSpec.getPlans().entrySet()) {
             Plan plan = generator.generate(entry.getValue(), entry.getKey(), serviceSpec.getPods());
             Assert.assertNotNull(plan);
-            Assert.assertEquals(2, plan.getChildren().size());
-            Assert.assertEquals(2, plan.getChildren().get(0).getChildren().size());
-            Assert.assertEquals(1, plan.getChildren().get(1).getChildren().size());
+            Assert.assertEquals(5, plan.getChildren().size());
+
+            Phase serverPhase = plan.getChildren().get(0);
+            Phase oncePhase = plan.getChildren().get(1);
+            Phase interleavePhase = plan.getChildren().get(2);
+            Phase fullCustomPhase = plan.getChildren().get(3);
+            Phase partialCustomPhase = plan.getChildren().get(4);
+
+            validatePhase(
+                    serverPhase,
+                    Arrays.asList(
+                            Arrays.asList("server"),
+                            Arrays.asList("server")));
+
+            validatePhase(
+                    oncePhase,
+                    Arrays.asList(
+                            Arrays.asList("once"),
+                            Arrays.asList("once")));
+
+            validatePhase(
+                    interleavePhase,
+                    Arrays.asList(
+                            Arrays.asList("once"),
+                            Arrays.asList("server"),
+                            Arrays.asList("once"),
+                            Arrays.asList("server")));
+
+            validatePhase(
+                    fullCustomPhase,
+                    Arrays.asList(
+                            Arrays.asList("once"),
+                            Arrays.asList("server"),
+                            Arrays.asList("server"),
+                            Arrays.asList("once")));
+
+            validatePhase(
+                    partialCustomPhase,
+                    Arrays.asList(
+                            Arrays.asList("server"),
+                            Arrays.asList("once"),
+                            Arrays.asList("once"),
+                            Arrays.asList("server")));
```

Comment:
Maybe bump the pod size for this one, so that we can validate that the `default` step holds for all steps after the first?